### ISSUE TITLE
auto.arima() seems to not export parameters properly to worker nodes

### ISSUE THREAD

**Issue by CarstenCroonenbroeck** [robjhyndman/forecast#855]
Hi,

it seems to me that `auto.arima()` does not export parameters properly to its worker nodes if parallelization is to be used. For example,

```
Max.p <- 20
Max.d <- 3
Max.q <- 20
Max.P <- 20
Max.D <- 3
Max.Q <- 20
Max.Order <- Max.p + Max.q + Max.P + Max.Q

RetVal <- auto.arima(WWWusage, max.p = Max.p, max.d = Max.d, max.q = Max.q, max.P = Max.P, max.D = Max.D, max.Q = Max.Q, max.order = Max.Order, stepwise = FALSE, approximation = TRUE, parallel = TRUE, num.cores = 7)
```

returns an error `7 nodes produced errors; first error: Objekt 'Max.Order' nicht gefunden` (that's German, it means "Object 'Max.Order' not found").

Is that a bug?

Best

Carsten

**Comment by arid-mangoss** [comment]
Thanks for the easily reproducible example! Yes, that looks like a bug to me. It happens [here ](https://github.com/robjhyndman/forecast/pull/874/files#diff-9d75e9c7fb1d184f74470520a03cff68fb98eb3393719c5ddbe20e6aa8dce192L69), but **not because** the `clusterExport()` is commented out. I tested it with it uncommented and the issue persists. Someone tried fixing it by exporting the variable in #830, but this did not work, which is why I believe it was eventually commented out.

The original PR #830 does not fix it because the `clusterExport()` occurs within a function, the function that is passed `max.order` as an argument. `max.order` is accessible within the environment of this function, but not 'globally'. When you call `clusterExport()`, by default you are grabbing the 'global' environment, with all the global variables (see the help page for `clusterExport()`, the default is `clusterExport(cl = NULL, varlist, envir = .GlobalEnv)`). 

In order to grab the variables passed into the method and make them accessible to the nodes, we can set `envir=environment()` in `clusterExport()`, which will then use the environment of the function that all this is defined in, thus making it so it can find the variable `max.order` when passed into `clusterExport(cl, c("max.order"), envir=environment())`.

**Comment by mitchelloharawild** [comment]
Fixed in #874
